### DISCUSSION TITLE
Obtaining the percentage of a gene within a Seurat object

### DISCUSSION THREAD

**Discussion by pcitrano3** [satijalab/seurat#5780] [Q&A]
Hello. I have tried searching for code to help me answering this, but I still cannot seem to work any of the proposed solutions or find what I am looking for. Currently, I have created an object of neuronal cells. However, I am worried that there is glial contamination. Is there any way that I can specifically see the percentage of a specific gene in my object? For instance my current object is called NeuN. I want to be able to find the percentage of the gene Gfap within my object. Is this possible to do via code or is there another way I have to go about this? Thank you so much for any help!

**Comment by alikhuseynov** [comment]
Hi there, I think you can try `PercentageFeatureSet()` 
eg `object[["percent.Gfap"]] <- PercentageFeatureSet(object = object, pattern = "Gfap", assay = "RNA")` 
this will give you the percentage of UMI (raw) counts for `Gfap` gene per cell. Then you can check which cells have higher percentage of that gene.
eg to check which cells have > 50% of `Gfap` gene and related metadata of an object.
`dplyr::filter(object@meta.data, percent.Gfap > 50) %>% str`

Other useful functions would be these:
`FoldChange()`
`AverageExpression()`

Hope this helps!

**Reply by pcitrano3** [reply]
Thank you so much!
